File: core/src/main/java/com/sequenceiq/cloudbreak/controller/json/PlatformVariantsJson.java
package com.sequenceiq.cloudbreak.controller.json;

import java.util.Collection;
import java.util.Map;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

@JsonIgnoreProperties(ignoreUnknown = true)
public class PlatformVariantsJson implements JsonEntity {

    private Map<String, Collection<String>> platformToVariants;
    private Map<String, String> defaultVariants;

    public Map<String, Collection<String>> getPlatformToVariants() {
        return platformToVariants;
    }

    public void setPlatformToVariants(Map<String, Collection<String>> platformToVariants) {
        this.platformToVariants = platformToVariants;
    }

    public Map<String, String> getDefaultVariants() {
        return defaultVariants;
    }

    public void setDefaultVariants(Map<String, String> defaultVariants) {
        this.defaultVariants = defaultVariants;
    }
}

File: core-api/src/main/java/com/sequenceiq/cloudbreak/api/model/PlatformVariantsJson.kt
package com.sequenceiq.cloudbreak.api.model

import com.fasterxml.jackson.annotation.JsonIgnoreProperties

@JsonIgnoreProperties(ignoreUnknown = true)
class PlatformVariantsJson : JsonEntity {

    var platformToVariants: Map<String, Collection<String>>? = null
    var defaultVariants: Map<String, String>? = null
}
